### PR TITLE
Fix #6141: CSS50.dat is never loaded

### LINKED ISSUE #6141: CSS50.dat is never loaded
OS: Windows 10
Version: 15063.483
Commit/Build: 18791fa

Css50.dat is copied into data directory of RCT2 but it don't load.

**Steps to reproduce:**
1. Rename Css17.dat rom RCT1 to Css50.dat and copy it to RCT2 data folder. 
2. Start game and try to select Roller Coaster Tycoon 1 theme music.
3. Watch game information that you don't have Css50.dat.

### PR DESCRIPTION
**Pull request by woj1993**
First attempt on fixing Css50 loading.
Fixes https://github.com/OpenRCT2/OpenRCT2/issues/6141 and https://github.com/OpenRCT2/OpenRCT2/issues/6466

Need tests.
Tested variants:
CSS50 exist.
CSS50 don't exist and RCT1 patch not set.
Need testing:
CSS50 don't exist and RCT1 patch set.

### CODE REVIEW

**Comment by woj1993** [comment]
@Gymnasiast Can you help me a little. How to check if patch to RCT1 has ben set? (_env->GetDirectoryPath(DIRBASE::RCT1, DIRID::DATA)) will be NULL?

**Comment by IntelOrca** [comment]
@woj1993 It looks like you are using the PR and CIs as your means of compiling and testing your changes. While there is nothing wrong with this, it does steal CI time from our develop branch and other PRs. You should instead work on a branch in your fork, create an AppVeyor account and enable your fork for it. That way your CI time does not hold up the project's jobs.

**Comment by woj1993** [comment]
@IntelOrca Sorry. I have a problem. Can you help me? I dont exactly know how to check if RCT1 directory has been set.

**Comment by IntelOrca** [comment]
You can probably check if result from `_env->GetDirectoryPath(DIRBASE::RCT1, DIRID::DATA)` is empty. If it is then you can assume it is not set.

**Comment by woj1993** [comment]
@IntelOrca I tried this. It didn't work.
pseudo code
if (!((_env->GetDirectoryPath(DIRBASE::RCT1, DIRID::DATA)).empty())){
\\load CSS17 from rct1
}
else
{
load CSS50 from rct2.
}

**Comment by IntelOrca** [comment]
Do not pass a `DIRID`, just do `_env->GetDirectoryPath(DIRBASE::RCT1)`

**Comment by woj1993** [comment]
Wait I see my error!

**Comment by woj1993** [comment]
@IntelOrca It works on my computer (Tested with CSS50 and without). Can you test it to confim it? I would be grateful and thanks for help.

**Comment by janisozaur** [comment]
@woj1993 you should first fix compilation before asking us to test it.

File: src/openrct2/Context.cpp [hunk 1]
```diff
@@ -256,6 +257,10 @@ namespace OpenRCT2
             {
                 auto dataPath = _env->GetDirectoryPath(DIRBASE::RCT1, DIRID::DATA);
                 result = Path::Combine(dataPath, "css17.dat");
+                if (!(std::ifstream(result.c_str()).good())){
```
Comment (janisozaur) [inline src/openrct2/Context.cpp]: You should not open a stream on a file just to test its existence.

Comment (woj1993) [reply src/openrct2/Context.cpp]: Then how would you do it?

Comment (IntelOrca) [reply src/openrct2/Context.cpp]: We have a `File::Exists` helper, use that.

Comment (woj1993) [reply src/openrct2/Context.cpp]: Ok. Thx @IntelOrca.

**Comment by woj1993** [comment]
@IntelOrca Sorry for problems but it fails. What I am doing wrong?

**Comment by woj1993** [comment]
@IntelOrca Please help!

**Comment by woj1993** [comment]
Squashed it into one commit.
@janisozaur Do you approve these changes?
@Gymnasiast It is not WIP anymore.

File: src/openrct2/Context.cpp [hunk 2]
```diff
@@ -254,8 +254,16 @@ namespace OpenRCT2
             std::string result;
             if (pathId == PATH_ID_CSS50)
             {
-                auto dataPath = _env->GetDirectoryPath(DIRBASE::RCT1, DIRID::DATA);
-                result = Path::Combine(dataPath, "css17.dat");
+                if (!(_env->GetDirectoryPath(DIRBASE::RCT1).empty()))
```
Comment (janisozaur) [inline src/openrct2/Context.cpp]: one set of parentheses is superfluous

**Comment by janisozaur** [comment]
The code looks better now, but I'd like someone who understand the issue signed off on this.

File: src/openrct2/Context.cpp (hunk 2 above)
Comment (woj1993) [reply src/openrct2/Context.cpp]: Which one?

**Comment by woj1993** [comment]
@janisozaur Thanks about review.
Issue was that there are 2 methods to load RCT1 menu music. One of them is renaming and moving RCT1 music file (CSS17.dat) to RCT2 data folder as CSS50.dat. It didn't get detected and loaded if you do this. This fixes it. There is only 1 case which I didn't test: if you have RTC1 patch set and don't have CSS50 in RCT2 folder.

**Comment by woj1993** [comment]
@janisozaur Can you appoint someone to look on this?

**Pull request merged**